### PR TITLE
Update: React-Hooks

### PR DESCRIPTION
**Pull request by mwaz**
A PR detailing why we need React Hooks and what they aim to solve.

### CODE REVIEW

File: react.md [hunk 1]
```diff
@@ -12,30 +12,34 @@ You can start with the official docs' [getting started](https://reactjs.org/docs
...
 <details>
-<summary>Why do we need hooks?</summary>
-
-If you know the answer to this question, please submit a pull request with the answer.
 
-</details>
+<summary>Why do we need hooks?</summary>
 
-_Waiting for response_
+Hooks are a way of letting the developer use State or other React Features without necessarily writing class components (sounds cool, right?!). Previously React required that to create state or set State of a component on needs to declare a class component where the state could be initialized and then would be later used in the class component. Using react hooks one can easily initialize state or any lifeclycle methods within functional components. An example would be setting initial state of counter in a functional component and incrementing the counter as a button is clicked.
```
Comment (emmabostian) [inline react.md]:
```suggestion
Hooks are a way of letting the developer use State or other React Features without necessarily writing class components (sounds cool, right?!). Previously React required that to create state or set State of a component on needs to declare a class component where the state could be initialized and then would be later used in the class component. Using React hooks one can easily initialize state or any lifecycle methods within functional components. An example would be setting initial state of counter in a functional component and incrementing the counter as a button is clicked. 
```

**Review by emmabostian — COMMENTED** [review]
Just a couple minor spelling errors! Thanks!

**Comment by mwaz** [comment]
@emmawedekind updated!

**Review by emmabostian — APPROVED** [review]
Woohoo! Thanks a bunch!

**Pull request merged**